llvm/test/CodeGen/DirectX/dot2add.ll

```diff
@@ -0,0 +1,8 @@
+; RUN: opt -S -dxil-op-lower -mtriple=dxil-pc-shadermodel6.3-compute %s | FileCheck %s
+
+define noundef float @dot2add_simple(<2 x half> noundef %a, <2 x half> noundef %b, float %c) {
+entry:
+; CHECK: call float @dx.op.dot2AddHalf(i32 162, float %c, half %0, half %1, half %2, half %3)
+  %ret = call float @llvm.dx.dot2add(<2 x half> %a, <2 x half> %b, float %c)

```

Comment:
Should we add half to the intrinsic name since the dxilOp has half in its  name?  This isn't a suggestion to change anything just curious.

Reply:
https://github.com/llvm/wg-hlsl/blob/main/proposals/0014-consistent-naming-for-dx-intrinsics.md is the official recommendation for stuff like this but it doesn't seem to comment on this one.

Does it make sense for there to ever be a "float" dot2add? If so, then I'd lean towards adding half to the name. But if this is the only spelling that makes sense then it's probably fine to drop it since the types are all included in the call

Reply:
Thank you @V-FEXrt for pointing the official documentation. My one cent is also towards not having half in the name as there is only 1 possible overload. At the same time, I am fine with either names.